### DISCUSSION TITLE
temporal group average: custom frequencies

### DISCUSSION THREAD

**Discussion by arfriedman** [xCDAT/xcdat#413] [2. Questions] [answered]
Hi all,

I was wondering if it is possible to create [temporal group averages](https://xcdat.readthedocs.io/en/latest/generated/xarray.Dataset.temporal.group_average.html) with custom daily or monthly frequencies -- for example, binned 5-day (pentad) or 5-month averages? 

Thanks much,
Andrew

**Comment by pochedls** [comment]
It is possible to create [custom seasons](https://xcdat.readthedocs.io/en/stable/generated/xarray.Dataset.temporal.group_average.html), which I think could handle your 5-month average example. I don't think there is any built-in functionality to create pentads – I'm not sure how hard this would be to implement. It could be a candidate for a feature request or a pull request.

**Comment by lee1043** [comment, marked as answer]
@arfriedman thank you for your interest in using xCDAT! 



1. If your input is daily time series and you are getting 5-day running average, I think xarray's [`rolling`](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.rolling.html) could be used ([example](https://stackoverflow.com/questions/45992983/python-xarray-rolling-mean-example)). 

2. However, when your input is monthly and you take 5-month average by using the xarray's [`rolling`](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.rolling.html), I guess the 5-month average is mean of 5 monthly time steps without considering weighting for different length of included months. 

3. When your input is daily and you are getting 5 month running average, I think separated steps are needed for 1) getting monthly average from daily, 2) getting 5 month running average from monthly mean. 

I agree to @pochedls that we don't have a built-in function for above 2 and 3 cases.

**Comment by arfriedman** [comment]
Thank you @lee1043 and @pochedls (I trid to accept both of your answers but it appears I can only select one).
